libs/langchain/langchain/schema/runnable/history.py

```diff
@@ -0,0 +1,254 @@
+from __future__ import annotations
+
+import asyncio
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Dict,
+    List,
+    Optional,
+    Sequence,
+    Type,
+    Union,
+)
+
+from langchain.load import load
+from langchain.pydantic_v1 import BaseModel, create_model
+from langchain.schema.chat_history import BaseChatMessageHistory
+from langchain.schema.runnable.base import Runnable, RunnableBindingBase, RunnableLambda
+from langchain.schema.runnable.config import RunnableConfig
+from langchain.schema.runnable.passthrough import RunnablePassthrough
+from langchain.schema.runnable.utils import (
+    ConfigurableFieldSpec,
+    get_unique_config_specs,
+)
+
+if TYPE_CHECKING:
+    from langchain.callbacks.tracers.schemas import Run
+    from langchain.schema.messages import BaseMessage
+
+MessagesOrDictWithMessages = Union[Sequence["BaseMessage"], Dict[str, Any]]
+
+
+class RunnableWithMessageHistory(RunnableBindingBase):
+    """
```

Comment:
The only way I will be happy is if we don't have a new line here. I'll probably be happy with a PS5 as well though (if i'm honest).